### PR TITLE
chore: add git-cliff for automated changelog generation

### PR DESCRIPTION
**Pull request by mederka**

### CODE REVIEW

**Comment by isidentical** [comment]
ATL: do we need to do anything locally, or is it going to generate everything on the GitHub Action run for the releases?

**Comment by mederka** [comment]
> ATL: do we need to do anything locally, or is it going to generate everything on the GitHub Action run for the releases?

Great question, @isidentical. No, we don't need to do anything locally. The should generate the changelog and put it in release notes automatically. We don't know if it will work for sure until we try releasing with this action. From my testing, I can see that the relevant changelog gets generated.

File: cliff.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+filter_unconventional = true
+# regex for preprocessing the commit messages
+commit_preprocessors = [
+    { pattern = '\((\w+\s)?#([0-9]+)\)', replace = "([#${2}](https://github.com/fal-ai/fal/pull/${2}))"},
+]
+# regex for parsing and grouping commits
+commit_parsers = [
+    { message = "^feat", group = "Features"},
+    { message = "^fix", group = "Bug Fixes"},
+    { message = "^doc", group = "Documentation"},
```
Comment (chamini2) [inline cliff.toml:33]: We have been doing `docs`. Should we add `docs` as a parser or `doc` works for both?

**Review by chamini2 — APPROVED** [review]

File: cliff.toml (hunk 1 above)
Comment (mederka) [reply cliff.toml:33]: `doc` works for both 👍

**Pull request merged**